components/cam/src/dynamics/se/dyn_comp.F90

```diff
@@ -322,6 +322,19 @@ subroutine dyn_init2(dyn_in)
        call write_grid_mapping(par, elem)
     end if
 
+    ! initialize dp3d from ps for FV physics grid
```

Comment:
Could you please explain why you init dp3d here? Generally the consideration is that elem%state%dp3d is not a valid field outside of homme dynamics tilmestep.

Reply:
dp3d is needed for a mass-weighting in the sub-cell averaging that is done in dyn_to_fv_phys() in fv_physics_coupling_mod.F90. It's normally not initialized until later, but for FV physigrd run the initial data is loaded onto the dynamics grid and needs to be converted to the physics grid to get the model going.

This was another thing that I took from Peter Lauritzen's branch, so I figured it was a valid form of dp. Is there a different dp variable I should use?

Reply:
This is not up to me to decide but it is a design assumption (which allows future design changes, like getting rid of dp3d in element) that dp3d is not valid outside of dynamics.

Reply:
ok, I'll leave it in unless I find an alternative. I figure it would be valid since I'm using it to integrate the dynamics fields.

Reply:
Could you simply build dp3d locally from ps_v?  I can't imagine ps_v will ever go away, but if the plan is to remove dp3d in the future this would protect this code from future design changes.  Similarly, dp3d is a state variable that is actually changed by dynamics and could potentially be wrong when you use it at the coupling layer.  For example, if the dynamics folks don't think of dp3d as anything other than a running tally of the pressure level thicknesses that is entirely under their control I could see a case where dp3d is inconsistent with the actual physics based pressure level thicknesses and it would be hard to catch since the inconsistentcy might be rather small.  That being said, at the dp and pd coupling layer the pressure level thicknesses should be consistent with the logic that builds the pressure levels from p0, ps_v, hyai, and hybi.

Reply:
ok, this is a good point. I think I can recalculate this in the FV coupling module. It seems odd to me that there wouldn't be a "dp" variable carried around by the dyn and phys state structures, but the re-calculation of do from ps shouldn't add significant overhead, so I'll just do that.

BTW, what does the "_v" imply in ps_v? seems like an odd naming convention to me, but I suspect I'm missing something

Reply:
You essentially want to borrow code like this:

```
        do k=1,nlev
           dp(:,:,k) = ( hvcoord%hyai(k+1) - hvcoord%hyai(k) )*hvcoord%ps0 + &
                ( hvcoord%hybi(k+1) - hvcoord%hybi(k))*elem%state%ps_v(:,:,np1)
enddo
```

or use the fact that the physics state variable `phys_state(lchnk)%ps(icol) = ps_tmp(ioff,ie)` is built from `ps_v` to locally calculate the pressure thickness.

As for what ps_v stands for, my best guess is the "surface pressure due to vapor" since it is found by adding up the total mass of the column of vapor above this level.  But I stress, this is a guess, I'm sure Oksana or Mark could clarify.

Reply:
well I can't use the physics state pressure, since I need dp to map from dyn to phys grids.

I agree that "vapor" makes sense.